### PR TITLE
Updated the Meshlets Sample Setup Instructions 

### PR DESCRIPTION
**Pull request by santorac**
## What does this PR do?

Updated the Meshlets setup instructions to address problems I encountered while setting it up myself.
This addresses the problems described in https://github.com/o3de/o3de/issues/11604

I also removed several unnecessary files from the ASV_GPU_and_CPU_Demo folder. These served as examples of what the files would look like after applying the steps in the Readme file, but there weren't actually referenced in the Readme, and they go stale quickly as changes continue to be made in AtomSampleViewer. It will be less confusing for users to just rely on the guidance in the Readme, and the folder only contains those files that the user should actually copy into AtomSampleViewer.

## How was this PR tested?

After these changes I was able to run the Meshlets sample in AtomSampleViewer and see partial results. 
Several issues were encountered with the sample which should be addressed as separate issues (see https://github.com/o3de/o3de/issues/11965):
1. There were gray depth clipping artifacts.
2. The reference model is not showing (see https://github.com/o3de/o3de/pull/9832) for the expected result.
3. AtomSampleViewer asserts on shutdown.

image
image

### CODE REVIEW

**Review by rgba16f — APPROVED** [review]

File: Gems/Meshlets/ASV_GPU_and_CPU_Demo/Readme.txt [hunk 1]
```diff
@@ -95,19 +95,23 @@ Epic Nanite 2021: https://quip-amazon.com/-/blob/RKb9AAL3zt8/G7A6Cxor0gvn_WPGltB
...
     "external_subdirectories": [
         "Gems/Meshlets",
         ...
-3. Run cmake again
-4. Compile meshoptimizer.lib as static library and add it to the Meshlets gem
-5. Build the ASV solution
-6. Run the standalone ASV and choose the Meshlets demo from the RPI demos sub-menu 
-7. When selecting a model, a secondary CPU meshlet model will be generated and 
+3. Compile meshoptimizer.lib as static library and add it to the Meshlets gem
+4. Update several build configuration files to connect the gem to AtomSampleViewer (ASV)
```
Comment (adiblev) [inline Gems/Meshlets/ASV_GPU_and_CPU_Demo/Readme.txt:109]: These changes in comments make it less obvious than what it was before.  I suggest leave this part as it was and point to the section that explain about the build configuration required (or the files that demonstrate it)

File: Gems/Meshlets/ASV_GPU_and_CPU_Demo/Readme.txt [hunk 2]
```diff
@@ -116,14 +120,26 @@ Adding the meshoptimizer library to the Gem
...
-Visual Studio you simply add the library file to the project).
+library and add it as part of Meshlets.Static project.
 Once this is done, the Gem should compile and link properly to allow you 
-to run the ASV sample 'Meshlets' (created with the MeshletsExampleComponent)
+to run the ASV sample 'Meshlets' (created with the MeshletsExampleComponent).
 
 The meshoptimizer library and source can be found in the following Github link:
 https://github.com/zeux/meshoptimizer
 
+Here are the specific steps to build and install the library, on Windows:
```
Comment (adiblev) [inline Gems/Meshlets/ASV_GPU_and_CPU_Demo/Readme.txt:130]: Great add - this should be referenced at the previous section as indicated.

File: Gems/Meshlets/ASV_GPU_and_CPU_Demo/Readme.txt [hunk 3]
```diff
@@ -163,18 +179,23 @@ is not the case, simply replace with the directory name of your active project.
...
+6. Copy the material type file 'DebugShaderPBR.materialtype' to the material types directory:
+    [o3de]\AtomSampleViewer\Materials\DebugShaderPBR.materialtype
     
-6. Complie meshoptimizer to a library, copy the compiled library (you can compile and place it under Gems\Meshlets\External\Lib) 
-    and add it to your Meshlets.static project.
-    
-Remark: at this point there is no need to copy the MainPipeline.pass that was provided 
-    as an example, since the meshlets parent pass is being injected into the pipeline by the 
-    MeshletsFeatureProcessor.
+7. Copy the material file 'debugshadermaterial_01.material' to the material directory:
```
Comment (adiblev) [inline Gems/Meshlets/ASV_GPU_and_CPU_Demo/Readme.txt:196]: I actually suggest now not doing it.  If you keep .shaders and .azsl in the Shaders section, and .materialtype and .material under Assets, it should be picked up by the AP.  Try it and once done, leave these files there and point to them.

**Review by adiblev — CHANGES_REQUESTED** [review]
1. I added some changes to the comments and location of things - one or two are important.
2. Not clear why were the cmake and connection to the sample manager removed.  They serve as a good reference (without need to be kept up to date), but indeed - perhaps adding a comment that they only serve as reference how to connect would be advised.

**Review by adiblev — COMMENTED** [review]
Another remark - it is not clear by the image if three copies of the object are rendered. I suspect that this is not the case and that the reference model is not rendered (broken material / shader?). If indeed three are rendered let's add an image presenting all three and comparison, and if not, let's figure out what broke.

File: Gems/Meshlets/ASV_GPU_and_CPU_Demo/Readme.txt [hunk 4]
```diff
@@ -116,14 +120,26 @@ Adding the meshoptimizer library to the Gem
...
+mkdir build
+cmake . -B build
+cmake --build build --config Release
+copy build\Release\meshoptimizer.lib [YourO3DEPath]\Gems\Meshlets\External\Lib
+
+Finally, in Visual Studio, right-click the Meshlets.Static project 
+    and click Add > Existing Item..., then select 
+    [YourO3DEPath]\Gems\Meshlets\External\Lib\meshoptimizer.lib
+    (Note that if you run cmake in O3DE again, this setting will be lost and you
+    will have to add meshoptimizer.lib to the Meshlets.Static project again).
```
Comment (AMZN-alexpete) [inline Gems/Meshlets/ASV_GPU_and_CPU_Demo/Readme.txt:142]: another possibility might be to use CMake's `FetchContent` to clone the repo automatically similar to how it is done in this PR https://github.com/o3de/o3de/pull/11626/files#diff-bf1793aef68dee31db8151fb3f89d21b51c97e43e4e7472c084fa06d9d904507

File: Gems/Meshlets/ASV_GPU_and_CPU_Demo/Readme.txt (hunk 1 above)
Comment (santorac) [reply Gems/Meshlets/ASV_GPU_and_CPU_Demo/Readme.txt:109]: I'm not sure what you mean, can you be more specific about why this is less obvious? 

I found the original instructions were hard to follow because this first section was high level and lacked detail, and it wasn't clear that this section was more like a summary. I was jumping around a lot trying to piece things together.

So I added a statement above "this is a summary of the steps you'll need to follow, see the additional sections below for details." Then the outline is some high level steps to give the reader an idea of what they'll be doing. Then they need to go through the remaining sections below to follow the details. My statement like "Update several build configuration files" is just meant to give the rough idea, and below there is a section called "Connecting the Gem to the project folder" where there's a bunch of details that cover this point. Perhaps it would help to rename these bullet points to exactly match the wording of the section titles below?

File: Gems/Meshlets/ASV_GPU_and_CPU_Demo/Readme.txt (hunk 2 above)
Comment (santorac) [reply Gems/Meshlets/ASV_GPU_and_CPU_Demo/Readme.txt:130]: That's what I meant by "This is a summary of the steps you'll need to follow, see the additional sections below for details."

File: Gems/Meshlets/ASV_GPU_and_CPU_Demo/Readme.txt (hunk 4 above)
Comment (santorac) [reply Gems/Meshlets/ASV_GPU_and_CPU_Demo/Readme.txt:142]: Yes, we'll definitely want to tie this into cmake at some point. The meshlets gem was left in a proof-of-concept state. I'm just trying to make minimal updates to the instructions to help readers follow a long, without getting too deep (I'm already outside my intended scope of work).

File: Gems/Meshlets/ASV_GPU_and_CPU_Demo/Readme.txt (hunk 3 above)
Comment (santorac) [reply Gems/Meshlets/ASV_GPU_and_CPU_Demo/Readme.txt:196]: I was just following along with the instructions that were already there. The instructions said to copy the .shader, .azsl, and .material but the .materialtype was missing. All I did was add .materialtype to the list.

**Comment by santorac** [comment]
>

**Comment by santorac** [comment]
> * Not clear why were the cmake and connection to the sample manager removed.  They serve as a good reference (without need to be kept up to date), but indeed - perhaps adding a comment that they only serve as reference how to connect would be advised.

I explained this in the PR description. IMO it was confusing to have these files included. I can add them back and update them to match the latest version of those files if you really want.

**Comment by santorac** [comment]
> Another remark - it is not clear by the image if three copies of the object are rendered. I suspect that this is not the case and that the reference model is not rendered (broken material / shader?). If indeed three are rendered let's add an image presenting all three and comparison, and if not, let's figure out what broke.

Only two models are showing. I opened a GitHub Issue about that problem and provided a link in the description. I consider this issue to be separate from the build setup problems I'm addressing here.

File: Gems/Meshlets/ASV_GPU_and_CPU_Demo/Readme.txt (hunk 4 above)
Comment (adiblev) [reply Gems/Meshlets/ASV_GPU_and_CPU_Demo/Readme.txt:142]: I think that the clear way would be to connect the meshoptimizer as a 3rdParty library but I never got to it.

**Review by adiblev — APPROVED** [review]

**Review by jeremyong-az — APPROVED** [review]
